### PR TITLE
KAFKA-9290: Update IQ related JavaDocs

### PR DESCRIPTION
**Pull request by highluck**
Update IQ related JavaDocs

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mjsax** [comment]
Thanks for the PR @highluck -- this overlaps with https://github.com/apache/kafka/pull/7848/ but the other PR seems to be abandoned? Hence, I would just more forward with your PR.

There are some other changes in #7848 (changing `KeyValueStore` to `ReadOnlyKeyValueStore` and adding missing generic types) -- also the class JavaDoc of `KTable` has a change in it. Can you include those changes in your PR?

\cc @miroswan who is the author of #7848 -- let us know if you are still interested in finishing your PR (we don't want to "take it away" from you).

**Comment by miroswan** [comment]
> Thanks for the PR @highluck -- this overlaps with #7848 but the other PR seems to be abandoned? Hence, I would just more forward with your PR.
> 
> There are some other changes in #7848 (changing `KeyValueStore` to `ReadOnlyKeyValueStore` and adding missing generic types) -- also the class JavaDoc of `KTable` has a change in it. Can you include those changes in your PR?
> 
> \cc @miroswan who is the author of #7848 -- let us know if you are still interested in finishing your PR (we don't want to "take it away" from you).

@mjsax I apologize. Please feel free to reassign.

**Comment by highluck** [comment]
@mjsax 
thanks for review

i'm update docs!

@miroswan 
thanks!

**Comment by highluck** [comment]
Retest this please.

**Comment by highluck** [comment]
@mjsax
Retest this please.

**Review by mjsax — COMMENTED** [review]
Sorry for the review delay. LGTM! Will merge after Jenkins passed.

**Comment by mjsax** [comment]
Retest this please

**Comment by mjsax** [comment]
Build failed with checkstyle error:
```
[ant:checkstyle] [ERROR] /home/jenkins/jenkins-slave/workspace/kafka-pr-jdk11-scala2.13/streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java:34:8: Unused import - org.apache.kafka.streams.processor.ProcessorContext. [UnusedImports]
11:59:28 [ant:checkstyle] [ERROR] /home/jenkins/jenkins-slave/workspace/kafka-pr-jdk11-scala2.13/streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java:163:21: '=' is not preceded with whitespace. [WhitespaceAround]
```

@highluck Can you update the PR?

**Comment by highluck** [comment]
Thank you for reivew
Yes ok!

**Comment by highluck** [comment]
@mjsax 

It was a hotfix!
#8515

File: streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java [hunk 1]
```diff
@@ -208,13 +208,13 @@
...
      * </pre>
-     * To query the local {@link KeyValueStore} it must be obtained via
+     * To query the local {@link ReadOnlyKeyValueStore} it must be obtained via
      * {@link KafkaStreams#store(StoreQueryParameters) KafkaStreams#store(...)}:
      * <pre>{@code
      * KafkaStreams streams = ...
-     * ReadOnlyKeyValueStore<String, Long> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<String, Long>keyValueStore());
-     * String key = "some-key";
-     * Long valueForKey = localStore.get(key); // key must be local (application state is shared over all running Kafka Streams instances)
+     * ReadOnlyKeyValueStore<K,ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedKeyValueStore());
```
Comment (ableegoldman) [inline streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java]: nit: insert space between generics (here and elsewhere)

**Comment by mjsax** [comment]
Ah. Thanks for pointing out the hotfix @highluck. Can you address Sophie's comment?

File: streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java (hunk 1 above)
Comment (highluck) [reply streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java]: ok! thanks :)

File: streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java [hunk 2]
```diff
@@ -212,7 +212,7 @@
      * {@link KafkaStreams#store(StoreQueryParameters) KafkaStreams#store(...)}:
      * <pre>{@code
      * KafkaStreams streams = ...
-     * ReadOnlyKeyValueStore<K,ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedKeyValueStore());
+     * ReadOnlyKeyValueStore<K,ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>> timestampedKeyValueStore());
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java]: I think @ableegoldman meant in the return type:
```
ReadOnlyKeyValueStore<K,ValueAndTimestamp<V>> localStore
// -> should be
ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore
```

The space you inserted should _not_ be there.

Comment (highluck) [reply streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java]: thanks i updated!

File: streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java [hunk 3]
```diff
@@ -212,7 +212,7 @@
      * {@link KafkaStreams#store(StoreQueryParameters) KafkaStreams#store(...)}:
      * <pre>{@code
      * KafkaStreams streams = ...
-     * ReadOnlyKeyValueStore<K,ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>> timestampedKeyValueStore());
+     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>> timestampedKeyValueStore());
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java [hunk 4]
```diff
@@ -397,7 +397,7 @@
...
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java [hunk 5]
```diff
@@ -439,7 +439,7 @@
...
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java [hunk 6]
```diff
@@ -116,7 +116,7 @@
      * <pre>{@code
      * KafkaStreams streams = ... // counting words
      * String queryableStoreName = "storeName"; // the store name should be the name of the store as defined by the Materialized instance
-     * ReadOnlyKeyValueStore<K,ValueAndTimestamp<Long>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<Long>> timestampedKeyValueStore());
+     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<Long>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<Long>> timestampedKeyValueStore());
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<Long>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<Long>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java [hunk 7]
```diff
@@ -161,7 +161,7 @@
...
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<Long>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<Long>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java [hunk 8]
```diff
@@ -263,7 +263,7 @@
      * <pre>{@code
      * KafkaStreams streams = ... // compute sum
      * String queryableStoreName = "storeName" // the store name should be the name of the store as defined by the Materialized instance
-     * ReadOnlyKeyValueStore<K,ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>> timestampedKeyValueStore());
+     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>> timestampedKeyValueStore());
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java [hunk 9]
```diff
@@ -326,7 +326,7 @@
...
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java [hunk 10]
```diff
@@ -429,7 +429,7 @@
      * <pre>{@code
      * KafkaStreams streams = ... // some aggregation on value type double
      * String queryableStoreName = "storeName" // the store name should be the name of the store as defined by the Materialized instance
-     * ReadOnlyKeyValueStore<K,ValueAndTimestamp<VR>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<VR>> timestampedKeyValueStore());
+     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<VR>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<VR>> timestampedKeyValueStore());
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<VR>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<VR>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java [hunk 11]
```diff
@@ -487,7 +487,7 @@
      * <pre>{@code
      * KafkaStreams streams = ... // some aggregation on value type double
      * String queryableStoreName = "storeName" // the store name should be the name of the store as defined by the Materialized instance
-     * ReadOnlyKeyValueStore<K,ValueAndTimestamp<VR>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<String, ValueAndTimestamp<VR>> timestampedKeyValueStore());
+     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<VR>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<String, ValueAndTimestamp<VR>> timestampedKeyValueStore());
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<VR>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<String, ValueAndTimestamp<VR>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java [hunk 12]
```diff
@@ -134,7 +134,7 @@
      * {@link KafkaStreams#store(StoreQueryParameters) KafkaStreams#store(...)}:
      * <pre>{@code
      * KafkaStreams streams = ... // filtering words
-     * ReadOnlyKeyValueStore<K,ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>> timestampedKeyValueStore());
+     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>> timestampedKeyValueStore());
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java [hunk 13]
```diff
@@ -173,7 +173,7 @@
...
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java [hunk 14]
```diff
@@ -259,7 +259,7 @@
...
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java [hunk 15]
```diff
@@ -297,7 +297,7 @@
...
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java]:
```suggestion
     * ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedKeyValueStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedCogroupedKStream.java [hunk 16]
```diff
@@ -160,7 +160,7 @@
      * <pre>{@code
      * KafkaStreams streams = ... // counting words
      * Store queryableStoreName = ... // the queryableStoreName should be the name of the store as defined by the Materialized instance
-     * ReadOnlyWindowStore<K,ValueAndTimestamp<V>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>> timestampedWindowStore());
+     * ReadOnlyWindowStore<K, ValueAndTimestamp<V>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>> timestampedWindowStore());
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedCogroupedKStream.java]:
```suggestion
     * ReadOnlyWindowStore<K, ValueAndTimestamp<V>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedWindowStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedCogroupedKStream.java [hunk 17]
```diff
@@ -216,7 +216,7 @@
...
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedCogroupedKStream.java]:
```suggestion
     * ReadOnlyWindowStore<K, ValueAndTimestamp<V>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedWindowStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedKStream.java [hunk 18]
```diff
@@ -130,7 +130,7 @@
      * <pre>{@code
      * KafkaStreams streams = ... // counting words
      * Store queryableStoreName = ... // the queryableStoreName should be the name of the store as defined by the Materialized instance
-     * ReadOnlyWindowStore<K,ValueAndTimestamp<Long>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<Long>> timestampedWindowStore());
+     * ReadOnlyWindowStore<K, ValueAndTimestamp<Long>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<Long>> timestampedWindowStore());
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedKStream.java]:
```suggestion
     * ReadOnlyWindowStore<K, ValueAndTimestamp<Long>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<Long>>timestampedWindowStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedKStream.java [hunk 19]
```diff
@@ -178,7 +178,7 @@
...
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedKStream.java]:
```suggestion
     * ReadOnlyWindowStore<K, ValueAndTimestamp<Long>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<Long>>timestampedWindowStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedKStream.java [hunk 20]
```diff
@@ -326,7 +326,7 @@
      * <pre>{@code
      * KafkaStreams streams = ... // counting words
      * Store queryableStoreName = ... // the queryableStoreName should be the name of the store as defined by the Materialized instance
-     * ReadOnlyWindowStore<K,ValueAndTimestamp<VR>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<VR>> timestampedWindowStore());
+     * ReadOnlyWindowStore<K, ValueAndTimestamp<VR>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<VR>> timestampedWindowStore());
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedKStream.java]:
```suggestion
     * ReadOnlyWindowStore<K, ValueAndTimestamp<VR>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<VR>>timestampedWindowStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedKStream.java [hunk 21]
```diff
@@ -386,7 +386,7 @@
...
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedKStream.java]:
```suggestion
     * ReadOnlyWindowStore<K, ValueAndTimestamp<VR>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<VR>>timestampedWindowStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedKStream.java [hunk 22]
```diff
@@ -543,7 +543,7 @@
      * <pre>{@code
      * KafkaStreams streams = ... // counting words
      * Store queryableStoreName = ... // the queryableStoreName should be the name of the store as defined by the Materialized instance
-     * ReadOnlyWindowStore<K,ValueAndTimestamp<V>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>> timestampedWindowStore());
+     * ReadOnlyWindowStore<K, ValueAndTimestamp<V>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>> timestampedWindowStore());
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedKStream.java]:
```suggestion
     * ReadOnlyWindowStore<K, ValueAndTimestamp<V>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedWindowStore());
```

File: streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedKStream.java [hunk 23]
```diff
@@ -605,7 +605,7 @@
...
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedKStream.java]:
```suggestion
     * ReadOnlyWindowStore<K, ValueAndTimestamp<V>> localWindowStore = streams.store(queryableStoreName, QueryableStoreTypes.<K, ValueAndTimestamp<V>>timestampedWindowStore());
```

**Comment by mjsax** [comment]
I removed the blanks you inserted on the wrong place in a previous commit but not not remove in your latest commit.

Thanks for the PR! Merged to `trunk`.

**Comment by highluck** [comment]
@mjsax 
thanks!!

**Pull request merged**